src/test/java/technology/dice/dicefairlink/iterators/RandomisedCyclicIteratorTest.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (C) 2018 - present by Dice Technology Ltd.
+ *
+ * Please see distribution for license.
+ */
+package technology.dice.dicefairlink.iterators;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import java.util.concurrent.ThreadLocalRandom;
+import org.assertj.core.api.Assertions;
+import org.junit.Before;
+import org.junit.Test;
+import java.util.ArrayList;
+
+public class RandomisedCyclicIteratorTest {
+
+  private static final int MIN_SIZE = 10;
+  private static final int MAX_SIZE = 1_000;
+  private static final String ELEMENT_PREFIX = "TEST_";
+
+  private int numberOfElementsToTest = -1;
+
+  @Before
+  public void setUp() {
+    numberOfElementsToTest = ThreadLocalRandom.current().nextInt(MIN_SIZE, MAX_SIZE);
```

Comment:
I don't think this is helpful. Supposing a random value did cause a test to fail, there would be no way to reproduce (bar running again hundreds of times until it reoccurred). However, the idea of testing with multiple generated values (property-based testing) is good, but use a framework that does it for us. JUnit-Quickcheck is one example, though there will be others

Reply:
Please use parameterised tests instead of this random variance.

If it fails, then what ? We can't investigate because we don't know what was the result of this random.

Reply:
Added extra log message so it will be possible to trace in case of error.